src/swap/SwapScreen.tsx

```diff
@@ -90,6 +90,18 @@ function getNetworkFee(quote: QuoteResult | null, networkId?: NetworkId) {
   }
 }
 
+function useFeeCurrenciesWithBalances(networkId: NetworkId) {
+  const feeCurrencies = useSelector((state) => feeCurrenciesSelector(state, networkId))
+
+  const feeCurrenciesWithBalances = useMemo(() => {
+    return feeCurrencies.filter((feeCurrency) => {
+      return feeCurrency.balance.isGreaterThan(0)
+    })
+  }, [feeCurrencies])
+
+  return feeCurrenciesWithBalances
```

Comment:
what do you think about making this a selector instead? i can't think of an example now but this seems like it could be useful outside of swaps too. i know we've been creating selectors _and_ hook pairs and i'm not sure that i understand why, it seems like selectors are superior because they can be more easily reused in sagas and we don't have to worry about `useMemo`?

Reply:
Yes I'll turn it into a selector 👍

Reply:
The hook with `useMemo` is simpler for situations where it's not needed for sagas.
Also allows keeping the memoized values only for when the component is mounted.

Whereas memoized `createSelector` are global and never cleared.